capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -107,12 +120,86 @@ public long upvotePost(HttpServletRequest request) {
 
     List<Entity> posts = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
 
-    int postID = Integer.parseInt(request.getParameter("id"));
-    Entity postToUpvote = posts.get(postID);
+    long postID = Long.parseLong(request.getParameter("id"));
+    Entity postToUpvote = null;
+    for (Entity post : posts) {
+      long entityID = post.getKey().getId();
+      if (entityID == postID) {
+        postToUpvote = post;
+        break;
+      }
+    }
+
+    if (postToUpvote == null) {
+      return -1;
+    }
 
     long currentUpvotes = (long) postToUpvote.getProperty("upvotes");
     postToUpvote.setProperty("upvotes", ++currentUpvotes);
     datastore.put(postToUpvote);
     return currentUpvotes;
   }
+
+  /**
+   * Sorts a list of entities based on the sort type passed in and returns the sorted list. If the
+   * sort type does not exist, an empty list is returned.
+   */
+  public List<Entity> sortEntities(String sortType, List<Entity> entities) {
+    Map<String, Comparator<Entity>> sorters = initializeSorters();
+    Comparator<Entity> sortMethod = sorters.get(sortType);
+    if (sortMethod == null) {
+      return Arrays.asList();
+    }
+
+    Collections.sort(entities, sortMethod);
+    return entities;
+  }
+
+  /** Returns a Map of all comparators. */
+  private Map<String, Comparator<Entity>> initializeSorters() {
+    Map<String, Comparator<Entity>> sorters = new HashMap<>();
+    sorters.put("new", sortByNew());
```

Comment:
My hunch is that you should move this map into a field and have them refer to method references https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html.